### PR TITLE
Use acquire_elem_ref() method when accessing meta data store from readers.

### PR DESCRIPTION
**Pull request by toregge**
Use ConstArrayRef to pass view of meta data to saver.

@geirst : please review

### CODE REVIEW

File: searchcore/src/vespa/searchcore/proton/documentmetastore/lid_gid_key_comparator.cpp [hunk 1]
```diff
@@ -7,15 +7,15 @@ namespace proton::documentmetastore {
 LidGidKeyComparator::LidGidKeyComparator(const document::GlobalId &gid,
                                          const MetaDataStore &metaDataStore)
     : _gid(gid),
-      _metaDataStore(metaDataStore),
+      _metaDataView(&metaDataStore.acquire_elem_ref(0)),
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/documentmetastore/lid_gid_key_comparator.cpp:10]: Consider adding a function on MetaDataStore that returns a view (e.g. make_read_view returning vespalib::ConstArrayRef < RawDocumentMetaData >) to make the usage clearer.

File: searchcore/src/vespa/searchcore/proton/documentmetastore/lid_gid_key_comparator.cpp [hunk 2]
```diff
@@ -7,15 +7,15 @@ namespace proton::documentmetastore {
...
       _gidCompare()
 {
 }
 
 LidGidKeyComparator::LidGidKeyComparator(const RawDocumentMetaData &metaData,
                                          const MetaDataStore &metaDataStore)
     : _gid(metaData.getGid()),
-      _metaDataStore(metaDataStore),
+      _metaDataView(&metaDataStore.acquire_elem_ref(0)),
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/documentmetastore/lid_gid_key_comparator.cpp:18]: See comment above.

**Review by geirst — APPROVED** [review]
👍

File: searchcore/src/vespa/searchcore/proton/documentmetastore/lid_gid_key_comparator.cpp (hunk 1 above)
Comment (toregge) [reply searchcore/src/vespa/searchcore/proton/documentmetastore/lid_gid_key_comparator.cpp]: Done.

File: searchcore/src/vespa/searchcore/proton/documentmetastore/lid_gid_key_comparator.cpp (hunk 2 above)
Comment (toregge) [reply searchcore/src/vespa/searchcore/proton/documentmetastore/lid_gid_key_comparator.cpp]: Done.

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**